src/rdkafka_cgrp.c

```diff
@@ -3730,25 +3750,34 @@ void rd_kafka_cgrp_handle_SyncGroup (rd_kafka_cgrp_t *rkcg,
 
 
 rd_kafka_consumer_group_metadata_t *
-rd_kafka_consumer_group_metadata_new (const char *group_id) {
+rd_kafka_consumer_group_metadata_new (const char *group_id,
+                                      int32_t generation_id) {
         rd_kafka_consumer_group_metadata_t *cgmetadata;
 
         if (!group_id)
                 return NULL;
 
         cgmetadata = rd_calloc(1, sizeof(*cgmetadata));
         cgmetadata->group_id = rd_strdup(group_id);
+        cgmetadata->generation_id = generation_id;
 
         return cgmetadata;
 }
 
 rd_kafka_consumer_group_metadata_t *
 rd_kafka_consumer_group_metadata (rd_kafka_t *rk) {
+        int32_t generation_id = -1;
+
         if (rk->rk_type != RD_KAFKA_CONSUMER ||
             !rk->rk_conf.group_id_str)
                 return NULL;
 
-        return rd_kafka_consumer_group_metadata_new(rk->rk_conf.group_id_str);
+        if (rk->rk_cgrp)
+                generation_id = rk->rk_cgrp->rkcg_generation_id;
```

Comment:
rkcg_generation_id needs to be initialized to -1 in cgrp_init  (maybe it is, havent gotten there yet)

Reply:
you mean `rd_kafka_cgrp_new`? yes.